fox_scraper/fox_scraper.py

```diff
@@ -48,63 +72,54 @@ def __init__(self, site_address):
         :param http_address: url address to send a request
-        :return: an html text to be scrap from
+        :return: an html parse text to be scrap from (beutifulsoup object)
         """
         raw_data = requests.get(http_address)
-        return raw_data.text
+        return BeautifulSoup(raw_data.text, 'html.parser')
 
-    def get_soup_object(self, web_text):
-        """
-        :param web_text: html txt file to scrap
-        :return: beautiful soup objet for scraping
-        """
-        return BeautifulSoup(web_text, 'html.parser')
-
-    def scrap_fox_runner(self):
+    def runner(self):
         """
         one of the main function for this class.
         this function will activate the scraping of the web address revieced on contruction of the class
         :return: a list of items (Item class) of the entire catalog
         """
-        soup_main_categories = self.scrap_fox_main_categories()
+        soup_main_categories = self.get_main_categories()
         catalog = []
-        soup_sub_categories = []
-        main_cat_num = 0
-        for soup_main_category in soup_main_categories:
-            soup_sub_categories += list(self.scrap_fox_sub_categories(soup_main_category, main_cat_num))
-            catalog.append(self.scrap_fox_catalog(soup_sub_categories, main_cat_num))
-            main_cat_num += 1
+        for main_cat_num in range(len(soup_main_categories)): # for every main-category
+            soup_sub_categories = self.get_sub_categories(soup_main_categories[main_cat_num], main_cat_num)
+            catalog.append(self.get_full_catalog(soup_sub_categories, main_cat_num))
         return catalog
 
-    def scrap_fox_main_categories(self):
+    def get_main_categories(self):
         """
         this function will scrap all main catalog categories
         :return: a list of beutifulSoup objects
         """
-        soup_main_category = self.get_soup_object(self.get_web_content(self.site_address))
+        soup_main_category = self.get_web_content(self.site_address)
         main_categories = []
         for link in soup_main_category.find_all(class_='link padding_hf_v '):
             main_categories.append(link.get('href'))
             self.main_categories.append(link.get_text().strip())
         return main_categories
 
-    def scrap_fox_sub_categories(self, soup_category, category_num):
+    def get_sub_categories(self, soup_category, category_num):
         """
         this function will scrap all catalog per sub category
         :param soup_category: sub-category link from the site
         :param category_num: category number for item description
         :return: a list of catalog items per sub-category
         """
-        sub_categories = self.get_soup_object(self.get_web_content(str(soup_category)))
+        sub_categories = self.get_web_content(str(soup_category))
         sub_catalog = []
         self.sub_categories.append(category_num)
         self.sub_categories[category_num] = []
```

Comment:
you can replace this 2 lines with `self.sub_categories.append([])`, it will do the same thing.